src/main/java/bio/terra/service/filedata/flight/ingest/IngestFileValidateCloudPlatformStep.java

```diff
@@ -0,0 +1,59 @@
+package bio.terra.service.filedata.flight.ingest;
+
+import bio.terra.common.CloudPlatformWrapper;
+import bio.terra.common.exception.NotImplementedException;
+import bio.terra.model.CloudPlatform;
+import bio.terra.service.dataset.Dataset;
+import bio.terra.service.dataset.exception.IngestFailureException;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class IngestFileValidateCloudPlatformStep implements Step {
+
+    private final Logger logger = LoggerFactory.getLogger(IngestFileValidateCloudPlatformStep.class);
+    private final Dataset dataset;
+
+    public IngestFileValidateCloudPlatformStep(Dataset dataset) {
+        this.dataset = dataset;
+    }
+
+    @Override
+    public StepResult doStep(FlightContext context) throws InterruptedException {
+        var profileModel = dataset.getDatasetSummary().getDefaultBillingProfile();
+        var profileCloudPlatform = profileModel.getCloudPlatform();
+        var profileWrapper = CloudPlatformWrapper.of(profileCloudPlatform);
+
+        if (profileWrapper.isAzure()) {
```

Comment:
What are the pros/cons of doing this check here vs doing it in `FileIngestFlight`? We have other flights where we conditionally add or not a step based on the current platform. Why wouldn't we do a similar check for an error case?

Reply:
I was trying to keep as much logic contained in steps and out of flights as possible, since this isn't really logic branching. If that's not a paradigm we care about, I'm happy to move it.